### PR TITLE
DCS support for LIFCL

### PR DESCRIPTION
**Pull request by mkurc-ant**
This one adds the `DCS` bel to the architecture.

### CODE REVIEW

File: libprjoxide/prjoxide/src/bels.rs [hunk 1]
```diff
@@ -584,6 +584,25 @@ impl Bel {
...
+            pins: vec![
+                input!(&postfix, "CLK0", "Clock input port 0 (default)"),
+                input!(&postfix, "CLK1", "Clock input port 1"),
+                input!(&postfix, "SEL", "Input clock select"),
+                input!(&postfix, "SELFORCE", "Selects glitchless (0) or non-glitchless (1) behavior"),
+                output!(&postfix, "DCSOUT", "Clock output port"),
+            ],
+            rel_x: 0,
+            rel_y: 0,
+            z: 0
```
Comment (gatecat) [inline libprjoxide/prjoxide/src/bels.rs]: this should probably be `z: 4` so it doesn't overlap with the DCCs at the same tile?

Comment (mkurc-ant) [reply libprjoxide/prjoxide/src/bels.rs]: Ah, right. I missed that, thanks!

**Comment by mkurc-ant** [comment]
I think that now this PR includes everything needed for the `DCS`. I managed to make it work in nextpnr and I'll make a PR there once its ready.

**Comment by gatecat** [comment]
Thanks!

**Pull request merged**